src/Domain/FundRepository.php

```diff
@@ -156,17 +156,42 @@ protected function getNewFund(array $fundData): Fund
         Assertion::string($name);
 
         if ($fundData['type'] === Pledge::DISCRIMINATOR_VALUE) {
-            $fund = new Pledge(currencyCode: $currencyCode, name: $name);
+            $fund = new Pledge(currencyCode: $currencyCode, name: $name, salesforceId: Salesforce18Id::of($fundData['id']));
         } elseif ($fundData['type'] === 'championFund') {
-            $fund = new ChampionFund(currencyCode: $currencyCode, name: $name);
+            $fund = new ChampionFund(currencyCode: $currencyCode, name: $name, salesforceId: Salesforce18Id::of($fundData['id']));
         } else {
             throw new \UnexpectedValueException("Unknown fund type '{$fundData['type']}'");
         }
-        $fund->setSalesforceId($fundData['id']);
 
         return $fund;
     }
 
+    /**
+     * @return Fund[]
+     */
+    public function findForCampaignsClosedSince(DateTime $closedSinceDate): array
+    {
+        $now = new DateTime('now');
```

Comment:
I'd be happier passing this in for testability/flexibility instead of accessing the system clock directly here.

Reply:
Fair enough, I was mostly matching what the donations counterpart does without thinking. I don't think I will use it for tests yet but moved it to a parameter just in case. https://github.com/thebiggive/matchbot/pull/1161/commits/15f7b77c1497f33e421bf94bae10e35771a48bed